### ISSUE TITLE
Need help regarding inference

### ISSUE THREAD

**Issue by karthik1145** [KevinMusgrave/pytorch-metric-learning#544]
I have trained an Efficient Net using Arcface Loss. I have saved the model weights and optimizer weights for model and loss function. 
I want to do inference on multiple new datasets using the best checkpoint. I have tried the 'get_logits' function, but it doesnt work. Could you please help me wrt this?

**Comment by KevinMusgrave** [comment]
Can you explain what you mean by "it doesn't work" ?

**Comment by karthik1145** [comment]
- The results are very wierd, the argmax values of all the data points in 1 even though the data is balanced between class 0 and class 1.
- Do I have to save loss.state_dict() and load it in order to get the correct logits? I was saving only model.state_dict()

**Comment by KevinMusgrave** [comment]
Yeah you have to save and load `loss.state_dict()` because the loss function contains parameters.

**Comment by karthik1145** [comment]
Thank you @KevinMusgrave .

**Comment by carlloos** [comment]
Hello, first I would like to thank for the library Kevin, helped me a lot in a university project. 
I'm doing inference with a model trained with Tripletloss + knn and im facing the same problem as karthik.
Could you give me an example how to load the  `loss.state_dict()` so i can get the embeddings and then perform classification with sklearn's knn?
I was only saving and loading the model's state_dict and the knn , but the model wasnt performing well with new test data.

**Comment by KevinMusgrave** [comment]
The `loss.state_dict()` only applies for classification losses like ArcFaceLoss.

Tuple losses like TripletMarginLoss don't have any weights, so there is nothing to load.
